app/objects/c_c2.py

```diff
@@ -113,3 +113,65 @@ def valid_config(self):
         :return: True if config is valid, False if not
         """
         return False
+
+    def start(self, app):
+        """
+        Default c2 task creation, useful for Active-style c2. Passive style c2 channels should override this function
+        :param app:
+        :return:
+        """
+        loop = asyncio.get_event_loop()
```

Comment:
this is the only funky part of the PR, that passive requires an override and active can use the function as-is. nothing technically wrong with that - but I would look for a way to emphasize that outside of (just) the doc string. you may want to consider an actual abc interface that sticks out. or a decorator that can override or execute the contents.